### PR TITLE
[Support] Refactoring contact

### PR DESCRIPTION
**Pull request by GSadee**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes
| Deprecations? | no
| Fixed tickets | #3545, #1951
| License       | MIT
| Doc PR        | no

Renamed Contact component to Support and ContactBundle to SupportBundle

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
@GSadee Remember to update the UPGRADE and CHANGELOG documents.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ This chapter references the relevant changes done in 0.15 version.
 
 To get the diff between two versions, go to https://github.com/Sylius/Sylius/compare/v0.14.0...master
 
+ * feature [BC BREAK][Support] Refactoring Contact
```
Comment (michalmarcinkowski) [inline CHANGELOG.md]: Add PR reference (#3164).

File: UPGRADE.md [hunk 2]
```diff
@@ -288,6 +288,45 @@ $ find ./src -type f -exec sed -i 's/Sylius\\Bundle\\ShippingBundle\\Checker/Syl
 $ find ./src -type f -exec sed -i 's/Sylius\\Bundle\\ShippingBundle\\Resolver/Sylius\\Component\\Shipping\\Resolver/g' {} \;
 $ find ./src -type f -exec sed -i 's/Sylius\\Bundle\\ShippingBundle\\Processor/Sylius\\Component\\Shipping\\Processor/g' {} \;
 ```
+```
+
+### Support
```
Comment (michalmarcinkowski) [inline UPGRADE.md]: This should be added at the very beginning. Under "From 0.14 to 0.15"

File: src/Sylius/Bundle/SupportBundle/Form/Type/CategoryTranslationType.php [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 
-namespace Sylius\Bundle\ContactBundle\Form\Type;
+namespace Sylius\Bundle\SupportBundle\Form\Type;
 
 use Sylius\Bundle\ResourceBundle\Form\Type\AbstractResourceType;
 use Symfony\Component\Form\FormBuilderInterface;
 
 /**
- * Contact topic translation form type.
+ * Contact category translation form type.
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/SupportBundle/Form/Type/CategoryTranslationType.php]: Can be removed (obvious comment).

File: src/Sylius/Bundle/SupportBundle/Form/Type/CategoryType.php [hunk 4]
```diff
@@ -9,18 +9,18 @@
...
 
-namespace Sylius\Bundle\ContactBundle\Form\Type;
+namespace Sylius\Bundle\SupportBundle\Form\Type;
 
 use Sylius\Bundle\ResourceBundle\Form\Type\AbstractResourceType;
 use Symfony\Component\Form\FormBuilderInterface;
 
 /**
- * Sylius contact topic form type.
+ * Sylius support category form type.
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/SupportBundle/Form/Type/CategoryType.php]: Remove all obvious comments.

**Comment by michalmarcinkowski** [comment]
By the way tou can also remove all fluent interfaces.

File: CHANGELOG.md (hunk 1 above)
Comment (pjedrzejewski) [inline CHANGELOG.md]: :+1:

File: features/support/support_category_management.feature [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        And there are following support categories:
+            | title               |
+            | Order return        |
+            | Delivery            |
+            | Product information |
+        And I am logged in as administrator
+
+    Scenario: Browsing all support categories
+        Given I am on the dashboard page
+        When I follow "Configure Categories"
```
Comment (pamil) [inline features/support/support_category_management.feature]: "Configure categories"?

File: features/support/support_ticket_management.feature [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    Background:
+        Given there is default currency configured
+        And there is default channel configured
+        And there are following support categories:
+            | title               |
+            | Order return        |
+            | Delivery            |
+            | Product information |
+        And there are following support tickets:
+            | category               | firstName | lastName | email            | message                    |
```
Comment (pamil) [inline features/support/support_ticket_management.feature]: Too many spaces in this table.

File: src/Sylius/Bundle/PayumBundle/spec/Payum/Action/ExecuteSameRequestWithPaymentDetailsActionSpec.php [hunk 7]
```diff
@@ -31,7 +31,7 @@ function it_extends_payum_payment_aware_action()
         $this->shouldHaveType('Payum\Core\Action\PaymentAwareAction');
     }
 
-    function it_should_support_request(Generic $request, PaymentInterface $payment)
+    function it_should_support_ticket(Generic $request, PaymentInterface $payment)
```
Comment (pamil) [inline src/Sylius/Bundle/PayumBundle/spec/Payum/Action/ExecuteSameRequestWithPaymentDetailsActionSpec.php]: You shouldn't modify `PayumBundle`.

File: src/Sylius/Bundle/SupportBundle/Behat/SupportContext.php [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+    {
+        $manager = $this->getEntityManager();
+        $repository = $this->getRepository('support_ticket');
+        $categoryRepository = $this->getRepository('support_category');
+
+        foreach ($repository->findAll() as $supportTicket) {
+            $manager->remove($supportTicket);
+        }
+
+        $manager->flush();
```
Comment (pamil) [inline src/Sylius/Bundle/SupportBundle/Behat/SupportContext.php]: Do `$manager->clear();` after flushing. There may be an error if you will try to add support ticket with the same data as previously deleted.

File: src/Sylius/Bundle/SupportBundle/Behat/SupportContext.php [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+        foreach ($table->getHash() as $data) {
+            $newSupportTicket = $repository->createNew();
+            $newSupportTicket->setFirstName($data['firstName']);
```
Comment (pamil) [inline src/Sylius/Bundle/SupportBundle/Behat/SupportContext.php]: It should be "first name", scenarios are for business people :)

File: src/Sylius/Bundle/SupportBundle/Behat/SupportContext.php [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+            $newSupportTicket->setLastName($data['lastName']);
```
Comment (pamil) [inline src/Sylius/Bundle/SupportBundle/Behat/SupportContext.php]: Same here.

File: src/Sylius/Bundle/SupportBundle/Behat/SupportContext.php [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+    public function thereAreSupportCategories(TableNode $table)
+    {
+        $manager = $this->getEntityManager();
+        $repository = $this->getRepository('support_category');
+
+        foreach ($repository->findAll() as $supportCategory) {
+            $manager->remove($supportCategory);
+        }
+
+        $manager->flush();
```
Comment (pamil) [inline src/Sylius/Bundle/SupportBundle/Behat/SupportContext.php]: Same as at line 32.

File: src/Sylius/Bundle/SupportBundle/SyliusSupportBundle.php [hunk 12]
```diff
@@ -38,9 +38,9 @@ public static function getSupportedDrivers()
     protected function getModelInterfaces()
     {
         return array(
-            'Sylius\Component\Contact\Model\RequestInterface'          => 'sylius.model.contact_request.class',
-            'Sylius\Component\Contact\Model\TopicInterface'            => 'sylius.model.contact_topic.class',
-            'Sylius\Component\Contact\Model\TopicTranslationInterface' => 'sylius.model.contact_topic_translation.class',
+            'Sylius\Component\Support\Model\TicketInterface'          => 'sylius.model.support_ticket.class',
```
Comment (pamil) [inline src/Sylius/Bundle/SupportBundle/SyliusSupportBundle.php]: All `=>` should be either in the same column or just after keys.

**Comment by pamil** [comment]
Btw. rename your commit messages to stick with convention eg. "[Support] Foo bar"

**Comment by michalmarcinkowski** [comment]
Rebase and use step `Given store has default configuration` in your features.

**Pull request closed without merging**